setup.py

```diff
@@ -1,12 +1,16 @@
 from setuptools import setup, find_packages
 
+
 setup(
     name="yunchang",
     version="0.2",
     author="Jiarui Fang, Zilin Zhu, Yang Yu",
     url="https://github.com/feifeibear/long-context-attention",
     packages=find_packages(exclude=['test', 'benchmark']),
     install_requires=[
-        'flash-attn',
+        'ninja',
+        # ROCM support has been added to main branch since Jun [PR#1010](https://github.com/Dao-AILab/flash-attention/pull/1010);
+        # flash_attn-2.6.3+rocm62 for pytorch 2.3a
+        'flash_attn @ git+https://git@github.com/Dao-AILab/flash-attention.git',
```

Comment:
```python
from setuptools import setup, find_packages

setup(
    name="yunchang",
    version="0.3",
    author="Jiarui Fang, Zilin Zhu, Yang Yu",
    url="https://github.com/feifeibear/long-context-attention",
    packages=find_packages(exclude=['test', 'benchmark']),
    install_requires=[
        'flash-attn>=2.6.0',  # default install cuda version
    ],
    extras_require={
        'amd': ['flash_attn @ git+https://git@github.com/Dao-AILab/flash-attention.git'],  # install amd version
    },
)
```

   ```bash
   pip install yunchuang
   ```

   ```bash
   pip install yunchuang[amd]
   ```

Reply:
I love the "extra_require".

Also, I suggest to use ninja. Otherwise, compiling will take painfully 30 mins, upto 1 hrs depends on network and cpu configuration because  Custlass/CK implementation needs to compile all codes templates.

Can we keep ninja ?

Reply:
Of course, we can keep "ninja". The compilation time for flash_attn is indeed very long. 

Does the AMD version of flash_attn have to be compiled from source? 
Can it be installed via pip install?